File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int n = s1.length();
        if (s1.equals(s2))
            return true;
        int cnt = 0;
        int[] a = new int[26];
        int[] b = new int[26];

        for (int i = 0; i < n; i++) {
            if (s1.charAt(i) != s2.charAt(i))
                cnt++;
            int idx1 = s1.charAt(i) - 'a';
            int idx2 = s2.charAt(i) - 'a';
            a[idx1]++;
            b[idx2]++;
        }
        for (int i = 0; i < 26; i++) {
            if (a[i] != b[i])
                return false;
        }
        return cnt <= 2;
    }
}

File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
var areAlmostEqual = function(s1, s2) {
    let n = s1.length;
    const a = new Array(26).fill(0);
    const b = new Array(26).fill(0);
    let cnt = 0;
    for (let i = 0; i < n; i++) {
        let idx1 = s1.charCodeAt(i) - 97;
        let idx2 = s2.charCodeAt(i) - 97;
        if (s1[i] != s2[i])
            cnt++;
        a[idx1]++;
        b[idx2]++;
    }
    for (let i = 0; i < 26; i++) {
        if (a[i] != b[i])
            return false;
    }
    return cnt <= 2;
};